UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common.UnitTests/HealthCheck/AzureMessageQueueHealthCheckTest.cs

```diff
@@ -0,0 +1,70 @@
+using UKHO.ExchangeSetService.Common.HealthCheck;
+using UKHO.ExchangeSetService.Common.Helpers;
+using UKHO.ExchangeSetService.Common.Models.Enums;
+using UKHO.ExchangeSetService.Common.Storage;
+
+namespace UKHO.ExchangeSetService.Common.UnitTests.HealthCheck
+{
+    public class AzureMessageQueueHealthCheckTest
+    {
+        private IAzureMessageQueueHelper fakeAzureMessageQueueHelperClient;
+        private ISalesCatalogueStorageService fakeSalesCatalogueStorageService;
+        private IOptions<EssFulfilmentStorageConfiguration> fakeEssFulfilmentStorageConfiguration;
+        private ILogger<AzureMessageQueueHelper> fakeLogger;
+        private AzureMessageQueueHealthCheck fakeAzureMessageQueueHealthCheck;
+        private IAzureBlobStorageService fakeAzureBlobStorageService;
+
+        [SetUp]
+        public void Setup()
+        {
+            this.fakeAzureMessageQueueHelperClient = A.Fake<IAzureMessageQueueHelper>();
+            this.fakeLogger = A.Fake<ILogger<AzureMessageQueueHelper>>();
+            this.fakeSalesCatalogueStorageService = A.Fake<ISalesCatalogueStorageService>();
+            this.fakeAzureBlobStorageService = A.Fake<IAzureBlobStorageService>();
+
+            this.fakeEssFulfilmentStorageConfiguration = Options.Create(new EssFulfilmentStorageConfiguration()
+            { QueueName = "testessdevqueue", StorageAccountKey = "testaccountkey", StorageAccountName = "testessdevstorage", StorageContainerName = "testContainer", DynamicQueueName = "testDynamicQueue", ExchangeSetTypes= "test" });
+
+            fakeAzureMessageQueueHealthCheck = new AzureMessageQueueHealthCheck(fakeAzureMessageQueueHelperClient, fakeSalesCatalogueStorageService, fakeEssFulfilmentStorageConfiguration, fakeLogger, fakeAzureBlobStorageService);
+        }
+
+        private (string, string) GetStorageAccountConnectionStringAndContainerName()
+        {
+            string storageAccountConnectionString = "DefaultEndpointsProtocol = https; AccountName = testessdevstorage2; AccountKey =testaccountkey; EndpointSuffix = core.windows.net";
+            string containerName = "testContainer";
+            return (storageAccountConnectionString, containerName);
+        }
+
+        [Test]
+        public async Task WhenAzureMessageQueueExists_ThenAzureMessageQueueIsHealthy()
+        {
+            A.CallTo(() => fakeSalesCatalogueStorageService.GetStorageAccountConnectionString(string.Empty,string.Empty)).Returns(GetStorageAccountConnectionStringAndContainerName().Item1);
+            A.CallTo(() => fakeAzureMessageQueueHelperClient.CheckMessageQueueHealth(A<string>.Ignored, A<string>.Ignored)).Returns(new HealthCheckResult(HealthStatus.Healthy, "Azure message queue is healthy"));
+            A.CallTo(() => fakeAzureBlobStorageService.GetInstanceCountBasedOnExchangeSetType(A<ExchangeSetType>.Ignored)).Returns(1);
+
+            var response = await fakeAzureMessageQueueHealthCheck.CheckHealthAsync(new HealthCheckContext());
```

Comment:
Is this named correctly: fakeAzureMessageQueueHealthCheck?

This is the object we are testing which is instantiated as a "new AzureMessageQueueHealthCheck(......." using fake dependency's.

Naming it as a fake is a little misleading as it took me a while to work out what we were actually testing - a real object or a fake!

Reply:
Fixed.